presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/credentials/file/FileCredentialsModule.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc.credentials.file;
+
+import com.google.inject.Binder;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.airlift.configuration.ConfigurationFactory;
+import io.prestosql.plugin.jdbc.credentials.CredentialProvider;
+import io.prestosql.plugin.jdbc.credentials.CredentialsConfig;
+import io.prestosql.plugin.jdbc.credentials.StaticCredentialProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.google.inject.Scopes.SINGLETON;
+import static io.airlift.configuration.ConfigurationLoader.loadPropertiesFrom;
+
+public class FileCredentialsModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        FileCredentialsConfig config = buildConfigObject(FileCredentialsConfig.class);
```

Comment:
Rather than reading the config inside the module, create a new `FileCredentialProvider` that has the `FileCredentialsConfig` injected.